src/openeo_gfmap/fetching/commons.py

```diff
@@ -76,19 +79,53 @@ def filter_condition(band_name, _):
     )
 
 
+def _load_collection_hybrid(
```

Comment:
This function does not seems to add a lot of functionality and might make it unnecessarily complex. As this code is only used in _load_collection if would simply create an ifthen there to keep it readable.

Reply:
I disagree on this point, as there is already 3 different places in the `_load_collection` method where a collection is fetched (already a if/else on the FetchType).

I think this helps to keep the `_load_collection` short (it's already a very long function) and abstracts the fact if we fetch from collection or from a stac

Reply:
Created issue #132